Documentation/ApiOverview/RequestLifeCycle/Typo3Request.rst

```diff
@@ -11,14 +11,136 @@ TYPO3 request object
 The TYPO3 request object is an implementation of the PSR-7 based
 `\Psr\Http\Message\ServerRequestInterface` containing TYPO3-specific attributes.
 
-The request object is passed to controllers. The attributes can be retrieved via
+..  seealso::
+    `PSR-7: HTTP message interfaces <https://www.php-fig.org/psr/psr-7/>`__
+
+.. contents::
+   :local:
+
+
+..  _getting-typo3-request-object:
+
+Getting the PSR-7 request object
+================================
+
+The PSR-7 based request object is available in most contexts. In some scenarios,
+like :ref:`PSR-15 middlewares <request-handling>` or backend module controllers,
+the PSR-7 base request object is given as argument to the called method.
+
+
+Extbase controller
+------------------
+
+..  versionadded:: 11.3
+
+The request object compatible with the PSR-7
+:php:`\Psr\Http\Message\ServerRequestInterface` is available in an
+:ref:`Extbase controller <extbase-action-controller>` via the class property
+:php:`$this->request`:
+
+..  code-block:: php
+    :caption: EXT:my_extension/Classes/Controller/MyController.php
+
+    public function myAction(): ServerResponseInterface
+    {
+        // ...
+
+        // Retrieve the language attribute via the request object
+        $language = $this->request->getAttribute('language');
+
+        // ...
+    }
+
+Prior to TYPO3 v11.3, a custom Extbase request object is available that does not
```

Comment:
Shall we still document that here? This information would belong in the 10.4 docs?

Reply:
When having an extension compatible with 10.4 and 11.5 it does matter (change is merged into 11.5). We can remove it later for the v12 docs.

Reply:
Mmh, then i would wirite what to do to be compatible with v11 and v10

Reply:
Agreed.